### PR TITLE
update the minimum version policy

### LINKED ISSUE #4179: Consider revising our minimum dependency version policy
Our [current policy](http://xarray.pydata.org/en/stable/installing.html#minimum-dependency-versions) is that xarray supports "the minor version (X.Y) initially published no more than N months ago" where N is:

- Python: 42 months (NEP 29)
- numpy: 24 months (NEP 29)
- pandas: 12 months
- scipy: 12 months
- sparse, pint and other libraries that rely on NEP-18 for integration: very latest available versions only,
- all other libraries: 6 months

I think this policy is too aggressive, particularly for pandas, SciPy and other libraries. Some of these projects can go 6+ months between minor releases. For example, version 2.3 of zarr is currently more than 6 months old. So if zarr released 2.4 *today* and xarray issued a new release *tomorrow*, and then our policy would dictate that we could ask users to upgrade to the new version.

In https://github.com/pydata/xarray/pull/4178, I misinterpreted our policy as supporting "the most recent minor version (X.Y) initially published more than N months ago". This version makes a bit more sense to me: users only need to upgrade dependencies at least every N months to use the latest xarray release.

I understand that NEP-29 chose its language intentionally, so that distributors know ahead of time when they can drop support for a Python or NumPy version. But this seems like a (very) poor fit for projects without regular releases. At the very least we should adjust the specific time windows.

I'll see if I can gain some understanding of the motivation for this particular language over on the NumPy tracker...

### PR DESCRIPTION
**Pull request by keewis**
As suggested in #4179, this updates the policy to "the most recent minor version (X.Y) initially published more than N months ago" or, slightly less correct but (for me) easier to understand: "the most recent minor version (X.Y) available N months ago".

This also reduces the policy months for python to 24 months and for numpy to 18 months.

- [x] Closes #4179
- [x] Passes `pre-commit run --all-files`
- [x] User visible changes (including notable bug fixes) are documented in `whats-new.rst`

### CODE REVIEW

File: doc/installing.rst [hunk 1]
```diff
@@ -111,9 +111,9 @@ dependencies:
   numpy >=1.17.
 - **all other libraries:** 12 months
 
-The above should be interpreted as *the minor version (X.Y) initially published no more
-than N months ago*. Patch versions (x.y.Z) are not pinned, and only the latest available
-at the moment of publishing the xarray release is guaranteed to work.
+The above should be interpreted as *the most recent minor version (X.Y) initially
+published more than N months ago*. Patch versions (x.y.Z) are not pinned, and only the
```
Comment (max-sixty) [inline doc/installing.rst:115]: How about this? (from [here](https://github.com/numpy/numpy/pull/14086#issuecomment-649762395))

```suggestion
This means the latest minor (X.Y) version from N months prior.
Patch versions (x.y.Z) are not pinned, and only the
```

Comment (dcherian) [reply doc/installing.rst:115]: The suggested change is very clear to me.

**Review by dcherian — APPROVED** [review]

File: doc/installing.rst (hunk 1 above)
Comment (keewis) [reply doc/installing.rst:115]: if anyone feels a particular wording is much easier to understand feel free to push to this branch

**Comment by keewis** [comment]
this should be ready for review / merging. There are also a few packages we are able to bump (most notably, `numpy` to `1.17` and `pandas` to `1.0`).

**Comment by max-sixty** [comment]
LGTM!

**Pull request merged**